src/main/java/io/cdap/plugin/http/sink/batch/HTTPSink.java

```diff
@@ -73,13 +76,21 @@ public void prepareRun(BatchSinkContext context) {
       .setFqn(config.getUrl()).build();
     LineageRecorder lineageRecorder = new LineageRecorder(context, asset);
     lineageRecorder.createExternalDataset(context.getInputSchema());
-    List<String> fields = inputSchema == null ?
-      Collections.emptyList() :
-      inputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    List<String> fields;
+    if (inputSchema == null) {
+      fields = Collections.emptyList();
+    } else {
+      fields = Objects.requireNonNull(Objects.requireNonNull(inputSchema).getFields()).stream()
+              .map(Schema.Field::getName).collect(Collectors.toList());
```

Comment:
please revert this `null` check. `inputSchema` `not null` should be validated in `configurePipeline` & `prepareRun`